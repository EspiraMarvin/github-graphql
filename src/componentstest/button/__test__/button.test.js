import React from 'react'
import ReactDOM from 'react-dom'
import Button from "../button";
import { render, cleanup } from '@testing-library/react';
import "@testing-library/jest-dom"
import renderer from "react-test-renderer"

afterEach(cleanup);

it('should render without crashing ',  () => {
    const div = document.createElement("div");
    ReactDOM.render(<Button></Button>, div)
    ReactDOM.unmountComponentAtNode(div)
});

it('should render button correctly', function () {
    const {getByTestId} = render(<Button label="this test button"></Button>)
    expect(getByTestId('button')).toHaveTextContent("this test button")
});

it('passing a different label', function () {
    const {getByTestId} = render(<Button label="save"></Button>)
    expect(getByTestId('button')).toHaveTextContent("save")
});

it('matches snapshot one', function () {
    const tree = renderer.create(<Button label="save"></Button>).toJSON();
    expect(tree).toMatchSnapshot();
});

it('matches snapshot two', function () {
    const tree = renderer.create(<Button label="dont save me"></Button>).toJSON();
    expect(tree).toMatchSnapshot();
});
