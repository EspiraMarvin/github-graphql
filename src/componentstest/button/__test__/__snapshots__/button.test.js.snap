// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot one 1`] = `
<div
  className="button-style"
  data-testid="button"
>
   hi 
  save
</div>
`;

exports[`matches snapshot two 1`] = `
<div
  className="button-style"
  data-testid="button"
>
   hi 
  dont save me
</div>
`;
